### PR TITLE
feat(argo-cd): Add support for external issuers in server Certificate resource

### PR DESCRIPTION
**Pull request by gregsidelinger**
Signed-off-by: Greg Sidelinger <sidelig@nationwide.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo).
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by gregsidelinger** [comment]
@mkilchhofer No idea why the chart-test check is not running but I did fix the issue in the docs that was detected last time it ran by running the helm-docs script.  Granted I did need to run jnorwood/helm-docs:v1.9.1 as latest was including comments in the docs and making a general mess of things.

File: charts/argo-cd/templates/argocd-server/certificate.yaml [hunk 1]
```diff
@@ -26,6 +26,7 @@ spec:
   duration: {{ . | quote }}
   {{- end }}
   issuerRef:
+    group: {{ .Values.server.certificate.issuer.group | quote }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-server/certificate.yaml:29]:
```suggestion
    {{- with .Values.server.certificate.issuer.group }}
    group: {{ . | quote }}
    {{- end }}
```

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Hi @gregsidelinger 

Let's make it optional as the CRD only requires `name` to be set.

> ```yaml
>                 issuerRef:
>                   description: IssuerRef is a reference to the issuer for this certificate. If the `kind` field is not set, or set to `Issuer`, an Issuer resource with the given name in the same namespace as the Certificate will be used. If the `kind` field is set to `ClusterIssuer`, a ClusterIssuer with the provided name will be used. The `name` field in this stanza is required at all times.
>                   type: object
>                   required:
>                     - name
>                   properties:
>                     group:
>                       description: Group of the resource being referred to.
>                       type: string
>                     kind:
>                       description: Kind of the resource being referred to.
>                       type: string
>                     name:
>                       description: Name of the resource being referred to.
>                       type: string
> ```
> Source: https://github.com/cert-manager/cert-manager/blob/master/deploy/crds/crd-certificates.yaml#L110-L124

Regarding README gen (helm-docs): Already opened an issue upstream https://github.com/norwoodj/helm-docs/issues/148

File: charts/argo-cd/templates/argocd-server/certificate.yaml [hunk 2]
```diff
@@ -26,6 +26,9 @@ spec:
   duration: {{ . | quote }}
   {{- end }}
   issuerRef:
+    {{- with .Values.server.certificate.issuer.group }}
+    group: {{ . | quote }}
+    {{-end }}
```
Comment (mkilchhofer) [inline charts/argo-cd/templates/argocd-server/certificate.yaml:31]:
```suggestion
    {{- end }}
```

**Review by mkilchhofer — APPROVED** [review]

File: charts/argo-cd/Chart.yaml [hunk 3]
```diff
@@ -21,4 +21,4 @@ dependencies:
     condition: redis-ha.enabled
 annotations:
   artifacthub.io/changes: |
-    - "[Changed]: Update Argo CD to v2.3.4"
+    - "[Added]: Added support for external issuers for cert-manager"
```
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:24]:
```suggestion
    - "[Added]: Add support for external issuers in server Certificate resource"
```

**Pull request merged**